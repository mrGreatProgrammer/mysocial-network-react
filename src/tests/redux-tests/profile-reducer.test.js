import profileReducer, {
  setUserProfile,
  updateUserStatusAC,
} from "../../redux/profile-reducer";

let profile = {
  id: 1,
  name: "myname",
  second_name: "secound",
  status: "double click to change your status",
};

let userStatus = "test test changed status";

let state = {
  userId: null,
  profile: {},
};

describe("--------- testing profile reducer SET_USER_PROFILE ---------", () => {
  it("right profile", () => {
    let action = setUserProfile(profile);
    let newState = profileReducer(state, action);
    expect(newState.profile).toEqual({
      id: 1,
      name: "myname",
      second_name: "secound",
      status: "double click to change your status",
    });
  });
});

describe("-------- testing profile reducer SET_USER_STATUS --------", () => {
  it("status didnt change", () => {
    let action = setUserProfile(profile);
    let newState = profileReducer(state, action);
    expect(newState.profile.status).toEqual(
      "double click to change your status"
    );
  });
  it("status changed", () => {
    let action = updateUserStatusAC(userStatus);
    let newState = profileReducer(state, action);
    expect(newState.profile.status).toEqual("test test changed status");
    expect(newState.profile.status).toMatchSnapshot()
  });
});
