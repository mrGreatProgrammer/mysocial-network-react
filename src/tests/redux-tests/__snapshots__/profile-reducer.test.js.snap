// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-------- testing profile reducer SET_USER_STATUS -------- status changed 1`] = `"test test changed status"`;
